noxfile.py

```diff
@@ -76,3 +104,38 @@ def run_examples(session: Session) -> None:
         for error in errors:
             print(f"- {error}")
         session.error(1)
+
+
+@nox.session(name="test:performance", python=False)
+def performance_tests(session: Session) -> None:
+    """Execute performance tests, assuming a DB already is ready
+
+    In the event of reasonable changes, the benchmark file should be updated
+    by running:
+        poetry run -- pytest ./test/performance/ --benchmark-sort=name
+        --benchmark-save=benchmark_performance
+        --benchmark-storage=file://test/performance/.benchmarks
+    Please save the newly created benchmark file over the existing one
+    at test/performance/.benchmarks/Linux-CPython-3.10-64bit/0001_benchmark_performance.json
+    and replace your local `"machine_info": { "key": "blah" }` with
+    `"machine_info": { }`.
+
+    Additionally, the value used in this nox session for `--benchmark-compare-fail`
+    should be re-evaluated and potentially updated.
+    """
+    test_path = _ROOT / "test/performance"
+    benchmark_path = test_path / ".benchmarks"
+
+    command = [
+        "pytest",
+        str(test_path),
+        "--benchmark-sort=name",
+        "--benchmark-compare=0001",
+        # the greatest stddev ~1s, so we set this at 1.25 of that for a robust estimate
```

Comment:
double-check: see this comment & below, but what kind of criteria would be beneficial here? should we split the import & exports or leave as is?